pkg/metadata/externalhost/externalhost.go

```diff
@@ -5,23 +5,34 @@
 
 package externalhost
 
-// externalHostCache maps hostname -> externalTags
-var externalHostCache = make(map[string]ExternalTags)
+// hostname -> ExternalTags
+type externalHost map[string]ExternalTags
 
-// AddExternalTags adds external tags for a specific host to the cache
-func AddExternalTags(hostname string, tags ExternalTags) {
-	externalHostCache[hostname] = tags
+// externalHostCache maps source_type -> externalHost
+var externalHostCache = make(map[string]externalHost)
+
+// SetExternalTags adds external tags for a specific host and source type
+// to the cache.
+func SetExternalTags(hostname, sourceType string, tags []string) {
+	_, found := externalHostCache[sourceType]
+	if !found {
+		externalHostCache[sourceType] = make(externalHost)
+	}
+
+	externalHostCache[sourceType][hostname] = ExternalTags{sourceType: tags}
```

Comment:
why not make `externalHostCache` simply a mapping from `hostname`s to `ExternalTags`, and update `externalHostCache[hostname][sourceType]` here? I don't understand the point of having the intermediate mapping sourceType -> externalHost

Reply:
In A5 the dict in the payload is expected to contain exactly one item, not sure whether this is a requirement or how likely would it be to happen having an hostname with tags from different source types but I tried to port the exact behaviour.
I can change it though, it would actually simplify the code.

Reply:
👍  let's merge as-is if it works. I had a quick look at our intake code and I think it should support multiple source_type per hostname in the payload, for example:
```
[
	["hostname1", {"vsphere": ["foo:val1", "bar:val1"], "openstack": ["foo2:val2", "bar2:val2"]}],
	["hostname2", {"vsphere": ["foo:val2", "bar:val2"]}]
]
```

But this would need some testing to confirm. Let's keep this for now.